app/ui/ScssButton/button.tsx

```diff
@@ -0,0 +1,27 @@
+import { Link, type LinkProps } from '@remix-run/react'
+import { type ButtonHTMLAttributes } from 'react'
+
+import './button.scss'
+
+export type ButtonProps = (
+  | (ButtonHTMLAttributes<HTMLButtonElement> & { as?: 'button' })
+  | (LinkProps & { as: 'link' })
+) & { primary?: boolean }
+
+export function ScssButton({ primary, className, children, ...props }: ButtonProps) {
+  const buttonClassName = `button ${primary ? 'button-primary' : 'button-secondary'} ${className}`
+
+  if (props.as === 'link') {
+    return (
+      <Link className={buttonClassName} {...props}>
```

Comment:
```suggestion
      <Link className={[styles.button, styles['button-primary']].join(' ')} {...props}>
```

Not sure it's the best way to add multiple classnames

Reply:
classNames lib handle this